sdk/storage/azure-storage-file-datalake/tests/test_directory.py

```diff
@@ -462,42 +548,64 @@ def progress_callback(resp):
+    def test_set_access_control_recursive_with_failures(self, **kwargs):
+        datalake_storage_account_name = kwargs.pop("datalake_storage_account_name")
+        datalake_storage_account_key = kwargs.pop("datalake_storage_account_key")
+        url = self.account_url(datalake_storage_account_name, 'dfs')
+        variables = kwargs.pop('variables', {})
+
         self._setUp(datalake_storage_account_name, datalake_storage_account_key)
-        if not self.is_playback():
-            return
+
         root_directory_client = self.dsc.get_file_system_client(self.file_system_name)._get_root_directory_client()
         root_directory_client.set_access_control(acl="user::--x,group::--x,other::--x")
 
-        # Using an AAD identity, create a directory to put files under that
+        # Create files and directories with provided owner except file3
+        test_guid = "5d56d308-df82-4266-ba63-ef1da3945873"
         directory_name = self._get_directory_reference()
+        directory_client = self.dsc.get_directory_client(self.file_system_name, directory_name)
+        directory_client.create_directory(owner=test_guid)
+        self.dsc.get_directory_client(self.file_system_name, directory_name + '/subdir1').create_directory(owner=test_guid, permissions='0777')
+        self.dsc.get_directory_client(self.file_system_name, directory_name + '/subdir2').create_directory(owner=test_guid, permissions='0777')
+        self.dsc.get_file_client(self.file_system_name, directory_name + '/subdir1/file1').create_file(owner=test_guid, permissions='0777')
+        self.dsc.get_file_client(self.file_system_name, directory_name + '/subdir2/file2').create_file(owner=test_guid, permissions='0777')
+        directory_client.get_file_client('file3').create_file()
+
+        # User delegation SAS with provided owner permissions
         token_credential = self.generate_oauth_token()
-        directory_client = DataLakeDirectoryClient(self.dsc.url, self.file_system_name, directory_name,
-                                                   credential=token_credential)
-        directory_client.create_directory()
-        num_sub_dirs = 5
-        num_file_per_sub_dir = 5
-        self._create_sub_directory_and_files(directory_client, num_sub_dirs, num_file_per_sub_dir)
-
-        # Create a file as super user
-        self.dsc.get_directory_client(self.file_system_name, directory_name).get_file_client("cannottouchthis") \
-            .create_file()
+        start_time = self.get_datetime_variable(variables, 'start_time', datetime.utcnow())
+        expiry_time = self.get_datetime_variable(variables, 'expiry_time', datetime.utcnow() + timedelta(hours=1))
+        owner_dsc = DataLakeServiceClient(url, credential=token_credential)
+        user_delegation_key = owner_dsc.get_user_delegation_key(start_time, expiry_time)
+        sas_token = self.generate_sas(generate_directory_sas,
```

Comment:
nit: I'm not really a fan of this style for the multi-line function call here. Would you mind changing to:
```py
sas_token = self.generate_sas(
    generate_directory_sas,
    datalake_storage_account_name,
    ...)
```

This will apply to all these `failed` tests if you decide to take this feedback.

Reply:
I can do that! I think I usually follow the convention you shared, but it must have slipped my mind as I hastily popped in the `self.generate_sas` when trying to get the recording to resolve (which ended up being a bigger ordeal bc of the sanitizers...)